### PR TITLE
rework s3 parameters

### PR DESCRIPTION
**Pull request by delgod**

### CODE REVIEW

File: lib/charms/mysql/v0/backups.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def is_unit_blocked(self) -> bool:
 import datetime
 import logging
 import pathlib
-from typing import Dict, List, Tuple
+from typing import Dict, List, Optional, Tuple
 
 from charms.data_platform_libs.v0.s3 import S3Requirer
 from charms.mysql.v0.mysql import (
```
Comment (shayancanonical) [inline lib/charms/mysql/v0/backups.py:54]: You would need to update the patchlib of all the libs modified in this PR: backups and s3_helpers

File: lib/charms/mysql/v0/mysql.py [hunk 2]
```diff
@@ -1401,9 +1397,13 @@ def execute_backup_commands(
             --storage=s3
             --parallel=10
             --md5
-            --s3-bucket={s3_bucket}
-            --s3-endpoint={s3_endpoint}
-            {s3_directory}
+            --storage=S3
```
Comment (shayancanonical) [inline lib/charms/mysql/v0/mysql.py:1400]: this option is already specified on line 1397

File: lib/charms/mysql/v0/backups.py (hunk 1 above)
Comment (delgod) [reply lib/charms/mysql/v0/backups.py:54]: Fixed https://github.com/canonical/mysql-operator/pull/204/commits/d55dd494463f7aac062d2645faeeead48c0848fc

File: lib/charms/mysql/v0/mysql.py [hunk 3]
```diff
@@ -67,7 +67,6 @@ def wait_until_mysql_connection(self) -> None:
 
 import json
 import logging
-import pathlib
```
Comment (shayancanonical) [inline lib/charms/mysql/v0/mysql.py:70]: The libpatch for this lib also needs to be updated

File: lib/charms/mysql/v0/mysql.py (hunk 2 above)
Comment (delgod) [reply lib/charms/mysql/v0/mysql.py:1399]: Fixed in https://github.com/canonical/mysql-operator/pull/204/commits/ff9bec18e87b75e09280c1ad93877d5efbc5a31f

File: lib/charms/mysql/v0/s3_helpers.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ def upload_content_to_s3(content: str, content_path: str, s3_parameters: Dict) -
 
 def _compile_backups_from_file_ids(
     metadata_ids: List[str], md5_ids: List[str], log_ids: List[str]
-) -> Tuple[str, str]:
```
Comment (shayancanonical) [inline lib/charms/mysql/v0/s3_helpers.py:77]: Libpatch needs to be updated

File: lib/charms/mysql/v0/mysql.py (hunk 3 above)
Comment (delgod) [reply lib/charms/mysql/v0/mysql.py:70]: fixed in https://github.com/canonical/mysql-operator/pull/204/commits/9a9f98955366f475ffbf6321a89b6a61260d9a50

File: lib/charms/mysql/v0/s3_helpers.py (hunk 4 above)
Comment (delgod) [reply lib/charms/mysql/v0/s3_helpers.py:77]: fixed in https://github.com/canonical/mysql-operator/pull/204/commits/6fd06a2660180876f4a968fe91c8e58213092650

**Review by shayancanonical — APPROVED** [review]

File: lib/charms/mysql/v0/backups.py [hunk 5]
```diff
@@ -133,17 +133,24 @@ def _retrieve_s3_parameters(self) -> Tuple[Dict, List[str]]:
...
-        s3_parameters.setdefault("region")
+        s3_parameters.setdefault("region", "")
         s3_parameters.setdefault("path", "")
+        s3_parameters.setdefault("s3-uri-style", "auto")
+        s3_parameters.setdefault("s3-api-version", "auto")
+
+        # Clean up extra slash symbols to avoid issues on 3rd-party storages
+        # like Ceph Object Gateway (radosgw)
+        s3_parameters["endpoint"] = s3_parameters["endpoint"].rstrip("/")
+        s3_parameters["path"] = s3_parameters["path"].strip("/")
```
Comment (taurus-forever) [inline lib/charms/mysql/v0/backups.py:144]: 1) Should we strip spaces from both sides (if any) for all params?
2) what about to remove "/" from bucket too? to be fully protected here?

```
juju config s3-integrator \
    endpoint="https://s3.amazonaws.com/" \
    bucket="postgresql-test-bucket-1/" \
    path="/postgresql-test/ " \
    region=" us-west-2"
```

**Review by taurus-forever — APPROVED** [review]
Nice! Thank you! One possible improvement to be fully protected.

File: lib/charms/mysql/v0/backups.py (hunk 5 above)
Comment (delgod) [reply lib/charms/mysql/v0/backups.py:144]: Leta do in separate PR

**Pull request merged**